lib/logstash/plugin.rb

```diff
@@ -59,6 +59,11 @@ def inspect
     end
   end
 
+  def dead_letter(event)
+    return unless event.is_a?(LogStash::Event)
+    LogStash::DeadLetterPostOffice << event
```

Comment:
I struggled a bit with this decision to either 
a) pass a DeadLetterDestination or DeadLetterPostOffice to each plugin
b) make the DLPO a singleton

Looking at how logger behaves, it's actually more of a singleton, since [plugin.rb](https://github.com/elastic/logstash/blob/master/lib/logstash/plugin.rb#L28) does `@logger = Cabin::Channel.get(LogStash)` and it's always the same channel identifier (LogStash). An approximate equivalent would be to do `@post_office = LogStash::DeadLetterPostOffice` in plugin.rb which doesn't seem that useful, imo.

The big driver question here really is: how fine grain will we want the dead letter destinations to be:
a) 1 per instance 
b) 1 per pipeline
c) 1 per plugin instance

Option d) is to have only one per instance which is: send dead letters to the  `dead_letter { ... }` section of the pipeline config.
This extra config section can then use conditionals, file output, stdout, etc